c/kastore.c

```diff
@@ -557,6 +579,35 @@ kastore_open(kastore_t *self, const char *filename, const char *mode, int flags)
     return ret;
 }
 
+int KAS_WARN_UNUSED
+kastore_openf(kastore_t *self, FILE *file, const char *mode, int flags)
+{
+    int ret = 0;
+
+    memset(self, 0, sizeof(*self));
+    if (strlen(mode) != 1) {
+        ret = KAS_ERR_BAD_MODE;
+        goto out;
+    }
+    if (strncmp(mode, "r", 1) == 0) {
+        self->mode = KAS_READ;
+    } else if (strncmp(mode, "w", 1) == 0) {
+        self->mode = KAS_WRITE;
```

Comment:
Not clear to me why append mode isn't allowed in openf - once the FILE is opened in update mode this should work find, right?

Reply:
The size of a store (and number of keys, etc.) is written in the header. This means that "append" has to write a new file, rather than just appending to the existing file. The current append code reads in the entire file, copies the data into a new store, and when the new store is written out (upon close), it has the appropriate header. I don't really see how this can be supported for an arbitrary FILE*.

Reply:
OK, simpler if we just don't support it. I'll update the docs to clarify.